lms/djangoapps/courseware/tests/test_course_info.py

```diff
@@ -227,18 +230,38 @@ def setUp(self):
             data="OOGIE BLOOGIE", display_name="updates"
         )
 
-    def test_info_title(self):
+        self.site = SiteFactory.create()
+
+    @ddt.data(
+        (False, True, True),
+        # (False, True, False),
+        # (True, True, True),
+        # (True, True, False),
+    )
+    @ddt.unpack
+    def test_info_title(self, invert_title, show_subtitle, show_org):
```

Comment:
It's easier to write ddt tests with the expected results laid out explicitly, instead of re-creating your logic in the test functions.

Also, there's an awesome context manager that can do all the site configuration magic: [with_site_configuration_context](https://github.com/edx/edx-platform/blob/af841336c7e39d634c238cd8a11c5a3a661aa9e2/openedx/core/djangoapps/site_configuration/tests/test_util.py#L46-L68).

```python
@attr(shard=1)
@ddt.ddt
class CourseInfoTitleTestCase(LoginEnrollmentTestCase, ModuleStoreTestCase):
    """
    Tests of the CourseInfo page title site configuration options.
    """
    def setUp(self):
        super(CourseInfoTitleTestCase, self).setUp()
        self.course = CourseFactory.create(
            org="HogwartZ",
            number="Potions_3",
            display_organization="HogwartsX",
            display_coursenumber="Potions101",
            display_name="Introduction to Potions"
        )

    @ddt.data(
        # Default site configuration shows course number, org, and display name as subtitle.
        (dict(),
         "Welcome to HogwartsX's Potions101!", "Introduction to Potions"),
        (dict(COURSE_HOMEPAGE_INVERT_TITLE=False,
              COURSE_HOMEPAGE_SHOW_SUBTITLE=True,
              COURSE_HOMEPAGE_SHOW_ORG=True),
         "Welcome to HogwartsX's Potions101!", "Introduction to Potions"),

        # Don't show org in title
        (dict(COURSE_HOMEPAGE_INVERT_TITLE=False,
              COURSE_HOMEPAGE_SHOW_SUBTITLE=True,
              COURSE_HOMEPAGE_SHOW_ORG=False),
         "Welcome to Potions101!", "Introduction to Potions"),

        # Hide subtitle and org
        (dict(COURSE_HOMEPAGE_INVERT_TITLE=False,
              COURSE_HOMEPAGE_SHOW_SUBTITLE=False,
              COURSE_HOMEPAGE_SHOW_ORG=False),
         "Welcome to Potions101!", None),

        # Show display name as title, hide subtitle and org.
        (dict(COURSE_HOMEPAGE_INVERT_TITLE=True,
              COURSE_HOMEPAGE_SHOW_SUBTITLE=False,
              COURSE_HOMEPAGE_SHOW_ORG=False),
         "Welcome to Introduction to Potions!", None),

        # Show display name as title with org, hide subtitle.
        (dict(COURSE_HOMEPAGE_INVERT_TITLE=True,
              COURSE_HOMEPAGE_SHOW_SUBTITLE=False,
              COURSE_HOMEPAGE_SHOW_ORG=True),
         "Welcome to HogwartsX's Introduction to Potions!", None),

        # Show display name as title, hide org, and show course number as subtitle.
        (dict(COURSE_HOMEPAGE_INVERT_TITLE=True,
              COURSE_HOMEPAGE_SHOW_SUBTITLE=True,
              COURSE_HOMEPAGE_SHOW_ORG=False),
         "Welcome to Introduction to Potions!", 'Potions101'),

        # Show display name as title with org, and show course number as subtitle.
        (dict(COURSE_HOMEPAGE_INVERT_TITLE=True,
              COURSE_HOMEPAGE_SHOW_SUBTITLE=True,
              COURSE_HOMEPAGE_SHOW_ORG=True),
         "Welcome to HogwartsX's Introduction to Potions!", 'Potions101'),
    )
    @ddt.unpack
    def test_info_title(self, site_config, expected_title, expected_subtitle):
        """
        Test the info page on a course without any display_* settings against
        one that does.
        """ 
        url = reverse('info', args=(unicode(self.course.id),))
        with with_site_configuration_context(configuration=site_config):
            response = self.client.get(url)
        
        content = pq(response.content)
    
        self.assertEqual(
            expected_title,
            content('.page-title').contents()[0].strip(),
        )
              
        if expected_subtitle is None:
            self.assertEqual(
                [],
                content('.page-subtitle'),
            )
        else: 
            self.assertEqual(
                expected_subtitle, 
                content('.page-subtitle').contents()[0].strip(),
            )
```

Your imports then change to:
```diff
--- a/lms/djangoapps/courseware/tests/test_course_info.py
+++ b/lms/djangoapps/courseware/tests/test_course_info.py
@@ -5,7 +5,6 @@ Test the course_info xblock
 import ddt
 import mock
 import pytest
-import json
 from django.conf import settings
 from django.core.urlresolvers import reverse
 from django.http import QueryDict
@@ -15,8 +14,8 @@ from ccx_keys.locator import CCXLocator
 from lms.djangoapps.ccx.tests.factories import CcxFactory
 from nose.plugins.attrib import attr
 from openedx.core.djangoapps.self_paced.models import SelfPacedConfiguration
+from openedx.core.djangoapps.site_configuration.tests.test_util import with_site_configuration_context
 from openedx.core.djangoapps.waffle_utils.testutils import WAFFLE_TABLES
-from openedx.core.djangoapps.site_configuration.tests.factories import SiteConfigurationFactory, SiteFactory
 from openedx.features.enterprise_support.tests.mixins.enterprise import EnterpriseTestConsentRequired
 from pyquery import PyQuery as pq
 from student.models import CourseEnrollment
```